File: rust/day-1/1_1.rs
// use std::env;
use std::fs;

fn main() {
    let file = "input.txt";
    let filename = format!("./{}", file);
    let binding = fs::read_to_string(filename).expect("Something went wrong reading the file");

    let contents = binding.trim().split("\n").collect::<Vec<&str>>();

    // println!("List: {:?}", contents);

    let mut list_1 = Vec::new();
    let mut list_2 = Vec::new();

    for i in 0..contents.len() {
        let t = contents[i].split("   ").collect::<Vec<&str>>();
        list_1.push(t[0].parse::<i32>().unwrap());
        list_2.push(t[1].parse::<i32>().unwrap());
        // println!("t: {:?}", t);
    }

    list_1.sort();
    list_2.sort();

    // println!("List 1: {:?}", list_1);
    // println!("List 2: {:?}", list_2);

    let mut result = 0;
    for i in 0..list_1.len() {
        result += (list_1[i] - list_2[i]).abs();
    }

    println!("Result: {:?}", result);
}

File: python/day-01/1_1.py
import re

inp = []
list_1 = []
list_2 = []
answer = 0

with open("input.txt", "r") as f:
  lines = f.readlines()
  for line in lines:
    inp.append(line.strip())

for i in inp:
  v = i.split("   ")
  list_1.append(v[0])
  list_2.append(v[1])

list_1.sort()
list_2.sort()

for i in range(len(list_1)):
  answer += abs(int(list_1[i]) - int(list_2[i]))

print(answer)
